### PR TITLE
[BEAM-10887] Timer clear Reapply timer clear

### PR DESCRIPTION
**Pull request by reuvenlax**
Resubmitting this PR without the unrelated changes to the Dataflow runner.

### CODE REVIEW

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java [hunk 1]
```diff
@@ -4851,6 +4892,348 @@ public void onTimer2(
...
+    //      final String timerId = "event-timer";
+    //
+    //      DoFn<KV<String, Integer>, Integer> fn =
+    //          new DoFn<KV<String, Integer>, Integer>() {
+    //
+    //            @TimerId(timerId)
+    //            private final TimerSpec spec = TimerSpecs.timer(TimeDomain.EVENT_TIME);
+    //
+    //            @ProcessElement
+    //            public void processElement(@TimerId(timerId) Timer timer, OutputReceiver<Integer>
```
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java:5213]: Did you mean to leave these large sections commented out?

**Review by lukecwik — APPROVED** [review]
LGTM

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java (hunk 1 above)
Comment (reuvenlax) [reply sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java]: No I didn't. thanks for catching!

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by tysonjh** [comment]
Is it possible that this caused the recent failure in `testTwoTimersSettingEachOtherWithCreateAsInputUnbounded ` https://issues.apache.org/jira/browse/BEAM-12589 ?

**Comment by reuvenlax** [comment]
When was the failure seen? The prior version of this PR that I rolled back
could definitely have caused this.

**Comment by reuvenlax** [comment]
Looking at the failure, I suspect this was caused by the bad PR, which I
have already rolled back.

**Comment by tysonjh** [comment]
OK thanks, I'll keep an eye on the next run.

**Comment by reuvenlax** [comment]
@tysonjh that test should be excluded currently from the Dataflow runner (the rolled-back PR accidentally removed that exclusion)

**Pull request merged**